kapitan/inputs/jsonnet.py

```diff
@@ -0,0 +1,79 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import ujson as json
+
+from kapitan.inputs.base import InputType, CompiledFile
+from kapitan.resources import search_imports, resource_callbacks
+from kapitan.utils import jsonnet_file, prune_empty
+
+logger = logging.getLogger(__name__)
+
+
+class Jsonnet(InputType):
+    def __init__(self, compile_path, search_paths, ref_controller):
+        super().__init__("jsonnet", compile_path, search_paths, ref_controller)
+
+    def compile_file(self, file_path, compile_path, ext_vars, **kwargs):
+        """
+        Write file_path (jsonnet evaluated) items as files to compile_path.
+        ext_vars will be passed as parameters to jsonnet_file()
+        kwargs:
+            output: default 'yaml', accepts 'json'
+            prune: default False, accepts True
+            reveal: default False, set to reveal refs on compile
+            target_name: default None, set to current target being compiled
+            indent: default 2
+        """
+        def _search_imports(cwd, imp):
+            return search_imports(cwd, imp, self.search_paths)
+
+        json_output = jsonnet_file(file_path, import_callback=_search_imports,
+                                   native_callbacks=resource_callbacks(self.search_paths),
+                                   ext_vars=ext_vars)
+        json_output = json.loads(json_output)
+
+        output = kwargs.get('output', 'yaml')
+        prune = kwargs.get('prune', False)
+        reveal = kwargs.get('reveal', False)
+        target_name = kwargs.get('target_name', None)
+        indent = kwargs.get('indent', 2)
+
+        if prune:
+            json_output = prune_empty(json_output)
+            logger.debug("Pruned output for: %s", file_path)
+
+        for item_key, item_value in json_output.items():
+            # write each item to disk
+            if output == 'json':
+                file_path = os.path.join(compile_path, '%s.%s' % (item_key, output))
+                with CompiledFile(file_path, self.ref_controller, mode="w", reveal=reveal, target_name=target_name,
+                                  indent=indent) as fp:
+                    fp.write_json(item_value)
+                    logger.debug("Wrote %s", file_path)
+            elif output == 'yaml':
+                file_path = os.path.join(compile_path, '%s.%s' % (item_key, "yml"))
+                with CompiledFile(file_path, self.ref_controller, mode="w", reveal=reveal, target_name=target_name,
+                                  indent=indent) as fp:
+                    fp.write_yaml(item_value)
+                    logger.debug("Wrote %s", file_path)
```

Comment:
Would it make sense to move this `logger.debug` call in a write_yaml super() function in CompiledFile ? Then you don't have to explicitly add it everywhere again and you always have a debug log.

Reply:
Yep

Reply:
done